src/lib/encrypt.h

```diff
@@ -49,7 +49,12 @@ static inline CK_RV encrypt_update (session_ctx *ctx, unsigned char *part, unsig
     return encrypt_update_op (ctx, NULL, part, part_len, encrypted_part, encrypted_part_len);
 }
 
-CK_RV encrypt_final_op (session_ctx *ctx, encrypt_op_data *supplied_opdata, unsigned char *last_encrypted_part, unsigned long *last_encrypted_part_len);
+CK_RV encrypt_final_ex (session_ctx *ctx, encrypt_op_data *supplied_opdata, unsigned char *last_encrypted_part, unsigned long *last_encrypted_part_len, bool is_oneshot);
+
+static inline CK_RV encrypt_final_op (session_ctx *ctx, encrypt_op_data *supplied_opdata, unsigned char *last_encrypted_part, unsigned long *last_encrypted_part_len) {
```

Comment:
Having both `encrypt_final_op()` and `encrypt_final_ex()` is confusing. Remove this function and call `encrypt_final_ex()` directly in `encrypt_final()`

Reply:
I thought I was matching sign, but sign dropped the op and just calls ex as you mention.